test/_init_tests.js

```diff
@@ -284,14 +284,28 @@ function warnForConsoleError() {
   setReportError(() => {});
   consoleErrorSandbox.stub(console, 'error').callsFake((...messages) => {
     const message = messages.join(' ');
+
+    // Match equal strings.
     if (expectedAsyncErrors.includes(message)) {
       expectedAsyncErrors.splice(expectedAsyncErrors.indexOf(message), 1);
       return;
-    } else {
-      // We're throwing an error. Clean up other expected errors since they will
-      // never appear.
-      expectedAsyncErrors = [];
     }
+
+    // Match regex.
```

Comment:
@rsimha 

Added partial/regex matching to `expectAsyncConsoleError` matching equivalent of `should.eventually.be.rejectedWith`.

Reply:
Thanks!

Reply:
This could clear more than one error if the regex is too broad. Not sure it's a real problem though.